### ISSUE TITLE
Reveal Scroll - fails on mobile

### ISSUE THREAD

**Issue by Ne-Ne** [zurb/foundation-sites#7548]
I have just checked full screen reveal, on mobile.

Test Case:

iOS - 9.2
iPhone 6
Safari

This was the result:

![image1](https://cloud.githubusercontent.com/assets/2177851/11809895/f8a2fd6a-a321-11e5-9e3c-89f79269d000.PNG)

You can scroll, when the intention is to lock the screen. I think iOS is not respect overflow hidden. Could this be included to support touch devices?

http://stackoverflow.com/questions/4770025/how-to-disable-scrolling-temporarily

**Comment by zurbrandon** [comment]
Oh man, that feels super dirty though. We'll take a look, but it feels like a really extreme option.

**Comment by zurbrandon** [comment]
@gerardkcohen @Owlbertz any thoughts on locking the scroll for iOS here?

**Comment by gerardkcohen** [comment]
Right, overflow:hidden on HTML will do it. A JS solution would be to trap
touch events to block "scrolling" of the viewport. Both have their issues,
with the CSS option my usual goto. Some gotchas for Firefox as well as
maintaining scroll position when overlay is dismissed. All as an FYI, I'm
sure you guys van work it out.

Let me know if you need anything else.

Gkc

**Comment by zurbrandon** [comment]
@gerardkcohen looks like overflow: hidden doesn't work on iOS in this instance though, which is a huge bummer.

**Comment by gerardkcohen** [comment]
@zurbrandon I use overflow:hidden here at work for that exact same thing with no issue. Interesting...

**Comment by gerardkcohen** [comment]
Oh, positioned:fixed on the overlay too helps.

Gkc

**Comment by zurbrandon** [comment]
Ah super odd, it's still scrolling. We may need to go another route.

**Comment by Ne-Ne** [comment]
I think that if you are using a full screen class it should lock the viewport... However this leads to issues with locking child elements and the overflow potential and so on.

Typically I thought chrome dev tools would help me out, however it does not inherit bugs or touch events.

@gerardkcohen, may I ask what device you are using?

**Comment by Ne-Ne** [comment]
@gerardkcohen It appears that the viewport meta tag affects the overflow property on the html & body. I don't think presuming all users will a surrounding class (for a overflow: hidden) the best idea.

What about a toggle function such as this: http://jsfiddle.net/3Ku26/

via - http://stackoverflow.com/questions/4770025/how-to-disable-scrolling-temporarily#answer-4770179

**Comment by gerardkcohen** [comment]
To solve this problem we decided to use a more simple solution that did not require JS, as it is less brittle. The JS solution proved problematic for us when we needed to support a largely unknown set of uses. We use the following when the overlay is visible:

```
html.overlayVisible,
html.overlayVisible body {
    height: 100%;
    overflow: hidden;
    user-select: none;
}
```

Our viewport meta tag is pretty basic: `<meta name="viewport" content="width=device-width, initial-scale=1">`

This has been pretty solid for us across all platforms/ browsers for the last 2 years.

HTH

**Comment by Ne-Ne** [comment]
Has anyone got ideas @zurbrandon, I put @gerardkcohen's fix into place using toggle events to a jQuery .on testing the body class and adding . overlayVisible to the HTML else removing it.

Granted I have a solution, but its not working solution going forward.

**Comment by Ne-Ne** [comment]
Being fairly new to Github pull requests. I have done the above.

I know one commit is similar to the other, but I couldn't see a delete commit on the web interface.

**Comment by DaSchTour** [comment]
I could also reproduce this in Chrome Desktop Mobile Simulator even with not fullscreen modal. Although I can't really test it, because the reveal is somehow to far at the bottom.
![google chrome 2016-01-27 16 33 15](https://cloud.githubusercontent.com/assets/1907266/12618367/d5ca072a-c513-11e5-99b3-2429e4c04cf4.png)

**Comment by zurbchris** [comment]
Hey everybody, please checkout [this branch](https://github.com/zurb/foundation-sites/tree/update-reveal) and take a peak. It resolves several issues with Reveal positioning, though we still have yet to tackle the overflow scrolling on iOS devices reliably.

**Comment by r8n5n** [comment]
I have been having issues with this on iOS using the new branch. I ended up wrapping all the main content (everything apart from the reveal divs) in a container div and then setting this to a width/height of 1px when the reveal is open. As far as I can tell so far it behaves as expected. 

Content structure:

`<body>

     <div id="main-content">
          
     </div>

     <div class="reveal full" id="exampleModal1" data-reveal>
         ......
     </div>

</body>`

And the CSS, using the _.is-reveal-open_ class that Foundation adds to _body_ when a reveal is open:

`    
//fix for iOS reveal scrolling
    .is-reveal-open{

        #main-content {
            position:absolute;
            top:auto;
            width:1px;
            height:1px;
            overflow:hidden;
        }
    }

    //smooth scrolling on iOS
    .reveal.full{
        -webkit-overflow-scrolling: touch;
    }
`

(Tested briefly in iOS9 on iPhone5/6 and iPad3)

**NB**
I tried doing this on the body tag but it did not work on the iPhone 6.

**Comment by pshoeg** [comment]
I had the same issue. I tried adding !important to the top value for reveal on small screens:

```
  @media screen and (max-width: 39.9375em) {
    .reveal {
      top: 0 !important;

```

It works like it's supposed to for me. I had to add some bottom padding as well, because the nav bar on the iPhone overlapped the content.

**Comment by kball** [comment]
I believe this was fixed, at least for iOS, in #8744.  This will be in the 6.2.2 release.

**Comment by jongar** [comment]
@kball the issue is still very much present in iOS Safari.

**Comment by kball** [comment]
@jongar can you confirm by still present you mean you're seeing this on 6.3?

**Comment by jongar** [comment]
@kball Yes. I also tested it with iPad Air and iPhone 6 Plus.

**Comment by joshbuchea** [comment]
@kball My organization is still seeing this issue with 6.3.

**Comment by kball** [comment]
OK, reopening this, 6.3.1 is under test now but let's make sure we tackle this for 6.3.2 and 6.4

**Comment by kball** [comment]
@jongar @joshbuchea Can either/both of you put some testcases up in codepen that reproduce? Template here: https://codepen.io/ncoden/pen/oYVpMe

**Comment by joshbuchea** [comment]
Steps to reproduce:

- Visit http://foundation.zurb.com/sites/docs/reveal.html#full-screen in iOS Safari
- Press "Click me for a full-screen modal" link
- Scroll

**Comment by gerardkcohen** [comment]
@kball @joshbuchea @jongar Pushed a quick fix #9763 that addresses the problem, but you should make sure it satisfies all appropriate use cases. HTH

**Comment by kball** [comment]
Fixed in #9763.

**Comment by DanielRuf** [comment]
It seems this produced a regression. The modal has a smaller height than the body but I can not scroll it unless I remove the `position: fixed´.
